src/software/ai/evaluation/keep_away.cpp

```diff
@@ -40,6 +40,15 @@ Point findKeepAwayTargetPoint(const World& world, const Pass& best_pass_so_far)
     // the region to which the optimization is (effectively) constrained to
     Circle keepaway_search_region(world.ball().position(), KEEPAWAY_SEARCH_CIRCLE_RADIUS);
 
+    // a reduced field rectangle to effectively constrain the optimization to field
+    // boundaries
+    const auto& field_bounds       = world.field().fieldLines();
+    Point reduced_bottom_left      = Point(field_bounds.xMin() + FIELD_SIZE_REDUCTION_M,
+                                      field_bounds.yMin() + FIELD_SIZE_REDUCTION_M);
+    Point reduced_top_right        = Point(field_bounds.xMax() - FIELD_SIZE_REDUCTION_M,
+                                    field_bounds.yMax() - FIELD_SIZE_REDUCTION_M);
+    Rectangle reduced_field_bounds = Rectangle(reduced_bottom_left, reduced_top_right);
```

Comment:
can we please have a simulated test for this case?

Reply:
yes

Reply:
Well, I meant a simulated test, i.e. https://drive.google.com/file/d/1mBpdEl8Lo85H85xXXMaUD4cLGH4YnxwQ/view?usp=sharing. In this video, the robot still leaves the field with the ball